sdk/core/core-auth/src/tokenCredential.ts

```diff
@@ -64,11 +47,29 @@ export interface AccessToken {
   expiresOnTimestamp: number;
 }
 
+/**
+ * Represents a credential capable of providing an authentication token.
+ */
+export interface TokenCredential {
+  /**
+   * Gets the token provided by this credential.
+   *
+   * This method is called automatically by Azure SDK client libraries. You may call this method
+   * directly, but you must also handle token caching and token refreshing.
+   *
+   * @param scopes The list of scopes for which the token will have access.
+   * @param options The options used to configure any requests this
+   *                TokenCredential implementation might make.
+   */
+  getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+}
+
 /**
  * Tests an object to determine whether it implements TokenCredential.
  *
  * @param credential The assumed TokenCredential to be tested.
  */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
```

Comment:
could we avoid this lint rule being disabled if we changed credential to be `unknown` and internally cast it to any if needed while doing the checks?

Reply:
Did this across all code.